recipes/mosquitto/2.x/conanfile.py

```diff
@@ -55,82 +76,101 @@ def _configure_cmake(self):
-        lib_suffix = "_static" if not self.options.shared else ""
-        # FIXME: there should be no namespace for CMake targets
-        self.cpp_info.components["libmosquitto"].name = "mosquitto"
-        self.cpp_info.components["libmosquitto"].libs = ["mosquitto" + lib_suffix]
-        self.cpp_info.components["libmosquitto"].names["pkgconfig"] = "libmosquitto"
-        if self.options.with_tls:
-            self.cpp_info.components["libmosquitto"].requires.append("openssl::openssl")
-            self.cpp_info.components["libmosquitto"].defines.append("WITH_TLS")
-        if self.settings.os == "Windows":
-            self.cpp_info.components["libmosquitto"].system_libs.append("ws2_32")
-            if self.options.with_tls:
-                self.cpp_info.components["libmosquitto"].system_libs.append("crypt32")
-        elif self.settings.os == "Linux":
-            self.cpp_info.components["libmosquitto"].system_libs.extend(["rt", "pthread", "dl"])
-
-        self.cpp_info.components["libmosquittopp"].name = "mosquittopp"
-        self.cpp_info.components["libmosquittopp"].libs = ["mosquittopp" + lib_suffix]
-        self.cpp_info.components["libmosquittopp"].requires = ["libmosquitto"]
-        self.cpp_info.components["libmosquittopp"].names["pkgconfig"] = "libmosquittopp"
-        if not self.options.shared:
-            self.cpp_info.components["libmosquitto"].defines.append("LIBMOSQUITTO_STATIC")
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        libsuffix = "" if self.options.shared else "_static"
+        self.cpp_info.components["libmosquitto"].names["pkg_config"] = "libmosquitto"
+        self.cpp_info.components["libmosquitto"].libs = ["mosquitto" + libsuffix]
+        self.cpp_info.components["libmosquitto"].requires = ["openssl::openssl"]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["libmosquitto"].system_libs = ["pthread", "m"]
+        elif self.settings.os == "Windows":
+            self.cpp_info.components["libmosquitto"].system_libs = ["ws2_32"]
+
+        if self.options.build_cpp:
+            self.cpp_info.components["libmosquittopp"].names["pkg_config"] = "libmosquittopp"
+            self.cpp_info.components["libmosquittopp"].libs = ["mosquittopp" + libsuffix]
+            self.cpp_info.components["libmosquittopp"].requires = ["libmosquitto"]
+            if self.settings.os == "Linux":
+                self.cpp_info.components["libmosquittopp"].system_libs = ["pthread", "m"]
+            elif self.settings.os == "Windows":
+                self.cpp_info.components["libmosquittopp"].system_libs = ["ws2_32"]
+
+        if self.options.broker:
+            self.cpp_info.components["broker"].libdirs = []
+            self.cpp_info.components["broker"].include_dirs = []
+            bin_path = os.path.join(self.package_folder, "bin")
```

Comment:
I am a little surprised to that broker does not require the base library, is this correct?

```suggestion
            self.cpp_info.components["broker"].requires = ["libmosquitto"]
            bin_path = os.path.join(self.package_folder, "bin")
```

Reply:
It's correct, broker does not require.